lib/sidekiq/throttled/patches/basic_fetch.rb

```diff
@@ -13,6 +14,29 @@ def apply!
           end
         end
 
+        # Timeout to sleep between fetch retries once at least
+        # timeout_after_attempts worth of throttled jobs are hit on a single
+        # queue. These balance scanning through a single queue for unthrottled
+        # jobs and not allowing lower priority queues to be completely starved
+        # by throttled jobs in higher queues
+        TIMEOUT_AFTER_ATTEMPTS = 100
+        TIMEOUT = 1
+
+        def initialize(cap)
+          super
+
+          sidekiq_version = Gem::Version.new(Sidekiq::VERSION)
+          if sidekiq_version < Gem::Version.new("7.0.0")
```

Comment:
Is this right?

Reply:
I think so, sidekiq 7 switched from a hash (where fetch works) to a custom class (where we need to use lookup). The tests past in this repo for sidekiq 6 and 7 with this